pwiz_tools/BiblioSpec/src/PSM.h

```diff
@@ -107,20 +107,49 @@ struct PSM{
     proteins.clear();
   };
 
+#define PRECURSOR_WITHOUT_MS2_SCAN "#PRECURSOR_ONLY#" // Precursor-only entries may not have an associated scan
+
+  static bool isPrecursorOnlyIdentifier(const std::string &str)
+  {
+      return boost::starts_with(str, PRECURSOR_WITHOUT_MS2_SCAN);
+  }
+
+  bool isPrecursorOnly()
+  {
+      return isPrecursorOnlyIdentifier(specName);
+  }
+  
+  void setPrecursorOnly()
+  {
+      specName = std::string(PRECURSOR_WITHOUT_MS2_SCAN) + "_"
+      + smallMolMetadata.moleculeName + "_" + smallMolMetadata.chemicalFormula + "_" + modifiedSeq + "_"
+      + smallMolMetadata.precursorAdduct + boost::lexical_cast<std::string>(charge);
+  }
+
   bool IsCompleteEnough() const
   {
       return (specKey >= 0 || specIndex >= 0 || !specName.empty()) && 
           (smallMolMetadata.IsCompleteEnough() ||
            (charge != 0 && !unmodSeq.empty()));
   }
+  
+  int idAsInteger() const
```

Comment:
I could not find this method being used anywhere. I probably would have named it "getHashCode".

Reply:
Yeah, that's cruft, thanks